launch/model_bundle.py

```diff
@@ -1,12 +1,176 @@
+import datetime
 from dataclasses import dataclass
-from typing import Any, Dict, List, Optional
+from enum import Enum
+from typing import Any, Dict, List, Optional, Union
 
 from dataclasses_json import Undefined, dataclass_json
+from pydantic import BaseModel, Field
+from typing_extensions import Literal
 
 # TODO(yi): These docstrings are currently perfunctory. I'm not sure we even want to expose most of these
 # fields. We need to overhaul our types :sadge:
 
 
+class ModelBundleFrameworkType(str, Enum):
```

Comment:
Long term, I think the ideal solution would be to have a `common` package (a la `botocore`) that has common definitions like this that are shared between server and client? But this is fine for now. cc @scaleapi/train-eng